### PR TITLE
Correct cross-references in documentation

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

Fix a few cross references in the documentation. Plus one error (cross-reference not working, but should be `output_dims` I assume.)

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/transforms.py [hunk 1]
```diff
@@ -1474,7 +1474,7 @@ def transform(self, values):
         Returns
         -------
         array
-            The output values as NumPy array of length :attr:`input_dims` or
+            The output values as NumPy array of length :attr:`output_dims` or
```
Comment (oscargus) [inline lib/matplotlib/transforms.py:1477]: There are more locations to possibly change this. I cannot get my head around if it should be `output_dims` or `input_dims` though...

Comment (jklymak) [reply lib/matplotlib/transforms.py:1477]: 99.9% sure `input_dims`!

Comment (tacaswell) [reply lib/matplotlib/transforms.py:1477]: I think "output_dims", but it is the case that in many cases they are both equal to 2 so it is correct by chance....

Comment (jklymak) [reply lib/matplotlib/transforms.py:1477]: Ooops, yes, I agree.  Sorry for being so certain yet so wrong.

**Comment by tacaswell** [comment]
https://67441-1385122-gh.circle-artifacts.com/0/doc/build/html/api/transformations.html#matplotlib.transforms.AffineBase.transform

The output dims did not xref corretly and the sentence should be self-consistent....

**Comment by tacaswell** [comment]
oh, there were three of them 🐑 Sorry I must have been looking at one you did not already change.

File: lib/matplotlib/transforms.py (hunk 1 above)
Comment (oscargus) [reply lib/matplotlib/transforms.py:1477]: OK! I've updated the remaining ones. Unfortunately, the subclasses documentation does not inherit the attributes. Not sure if that can be done in a better way than tell that all inherited attributes should be shown?

Anyway, it is now correct (assuming that `output_dims` is indeed correct), although the links don't work.

**Comment by QuLogic** [comment]
Are there entries for these in `missing-references.json` that can be removed?

**Comment by oscargus** [comment]
> Are there entries for these in `missing-references.json` that can be removed?

Yes, most of these (not `output_dims`) and some others can be removed.

I have plans to sort out more as I go along and make a proper rerun at some stage, but there are some fundamental things to sort out first (like getting inherited attributes and methods in so that they can be referenced and possibly sorting out the pyplot  issues). But I can of course remove these manually for now if that is preferred.

(Also, one should maybe convert all missing references in old release notes to strings. Like:
https://github.com/matplotlib/matplotlib/blob/6b2d3f29b81aaf29c73828957a47fc75a6ce375e/doc/missing-references.json#L69-L71

Possibly the same for private classes that do not have any documentation pages. Like:
https://github.com/matplotlib/matplotlib/blob/6b2d3f29b81aaf29c73828957a47fc75a6ce375e/doc/missing-references.json#L195-L197
or
https://github.com/matplotlib/matplotlib/blob/6b2d3f29b81aaf29c73828957a47fc75a6ce375e/doc/missing-references.json#L352-L354)

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
It's probably fine to just update them in bulk.

**Pull request merged**